test/e2e/egressip.go

```diff
@@ -871,6 +875,160 @@ spec:
+			}
+		}()
+
+		framework.Logf("Create the EgressIP configuration")
+		framework.RunKubectlOrDie("default", "create", "-f", egressIPYaml)
+
+		ginkgo.By("3. Check that the status is of length one and that it is assigned to node2 (pod2Node/egress1Node)")
+		statuses := verifyEgressIPStatusLengthEquals(1, nil)
+		if statuses[0].Node != egress1Node.name {
+			framework.Failf("Step 3. Check that the status is of length one and that it is assigned to node2 (pod2Node/egress1Node), failed")
+		}
+
+		ginkgo.By("4. Check connectivity from pod to an external container and verify that the srcIP is the expected egressIP from object1")
+		err = wait.PollImmediate(retryInterval, retryTimeout, targetExternalContainerAndTest(targetNode, pod1Name, podNamespace.Name, true, []string{egressIP1.String()}))
+		framework.ExpectNoError(err, "Step 4. Check connectivity from pod to an external container and verify that the srcIP is the expected egressIP from object1, failed: %v", err)
+
+		ginkgo.By("5. Create an EgressIP object2 with one egress IP defined")
+		// Assign the egress IP without conflicting with any node IP,
+		// the kind subnet is /16 or /64 so the following should be fine.
+		egressIP2 := dupIP(egressNodeIP)
+		egressIP2[len(egressIP2)-2]++
+		egressIP2[len(egressIP2)-1]++
+
+		var egressIPConfig2 = fmt.Sprintf(`apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+    name: ` + egressIPName2 + `
+spec:
+    egressIPs:
+    - ` + egressIP2.String() + `
+    podSelector:
+        matchLabels:
+            wants: egress
+    namespaceSelector:
+        matchLabels:
+            name: ` + f.Namespace.Name + `
+`)
+		if err := ioutil.WriteFile(egressIPYaml, []byte(egressIPConfig2), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		defer func() {
+			if err := os.Remove(egressIPYaml); err != nil {
+				framework.Logf("Unable to remove the CRD config from disk: %v", err)
+			}
+		}()
+
+		framework.Logf("Create the EgressIP configuration")
+		framework.RunKubectlOrDie("default", "create", "-f", egressIPYaml)
+
+		ginkgo.By("6. Check that the status is of length one and that it is assigned to node2 (egress1Node)")
+		egressIPs := egressIPs{}
+		egressIPStdout, err := framework.RunKubectl("default", "get", "eip", "-o", "json")
+		if err != nil {
+			framework.Failf("Error: failed to get the EgressIP object, err: %v", err)
+		}
+		json.Unmarshal([]byte(egressIPStdout), &egressIPs)
+		statusEIP1 := egressIPs.Items[0].Status.Items
+		statusEIP2 := egressIPs.Items[1].Status.Items
+		if statusEIP1[0].Node != egress1Node.name || statusEIP2[0].Node != egress1Node.name {
+			framework.Failf("Step 6. Check that the status is of length one and that it is assigned to node2 (egress1Node), failed")
+		}
+
+		// NOTE: We don't have a solid way to ensure ovn-db setup was not done for second object.
+		// TODO: Check logs/DB to ensure only one SNAT/reroute is present for the pod at a given time
```

Comment:
@trozet : do you want me to add this bit as well? To ensure we actually do an nbctl call into db to check there is only one pod SNAT?

Reply:
sure

Reply:
done.